stdlib/universe/hourSelection_test.go

```diff
@@ -0,0 +1,424 @@
+						{Label: "tag1", Type: flux.TString},
+					},
+					Data: [][]interface{}{
+						{execute.Time(3600000000000), "a", "b"},
+						{execute.Time(3600000000000 * 3), "a", "c"},
+						{execute.Time(3600000000000 * 5), "a", "b"},
+						{execute.Time(3600000000000 * 7), "a", "d"},
+					},
+				},
+			},
+			want: []*executetest.Table{{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "tag0", Type: flux.TString},
+					{Label: "tag1", Type: flux.TString},
+				},
+				Data: [][]interface{}(nil),
+			}},
+		},
+		{
+			name: "no group key -- outside range",
+			spec: &universe.HourSelectionProcedureSpec{Start: 25, Stop: 40},
+			data: []flux.Table{
+				&executetest.Table{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "tag0", Type: flux.TString},
+						{Label: "tag1", Type: flux.TString},
+					},
+					Data: [][]interface{}{
+						{execute.Time(3600000000000 * 6), "a", "b"},
+						{execute.Time(3600000000000 * 25), "a", "c"},
+						{execute.Time(3600000000000 * 33), "a", "b"},
+						{execute.Time(3600000000000 * 39), "a", "d"},
+					},
+				},
+			},
+			want: []*executetest.Table{{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "tag0", Type: flux.TString},
+					{Label: "tag1", Type: flux.TString},
+				},
+				Data: [][]interface{}(nil),
+			}},
+		},
+		{
+			name: "no group key -- multiple with same hour",
+			spec: &universe.HourSelectionProcedureSpec{Start: 6, Stop: 6},
+			data: []flux.Table{
+				&executetest.Table{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "tag0", Type: flux.TString},
+						{Label: "tag1", Type: flux.TString},
+					},
+					Data: [][]interface{}{
+						{execute.Time(3600000000000 * 6), "a", "b"},
+						{execute.Time(3600000100000 * 6), "a", "c"},
+					},
+				},
+			},
+			want: []*executetest.Table{{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "tag0", Type: flux.TString},
+					{Label: "tag1", Type: flux.TString},
+				},
+				Data: [][]interface{}{
+					{execute.Time(3600000000000 * 6), "a", "b"},
+					{execute.Time(3600000100000 * 6), "a", "c"},
+				},
+			}},
+		},
+		{
+			name: "with two group keys, multiple days",
+			spec: &universe.HourSelectionProcedureSpec{Start: 4, Stop: 6},
+			data: []flux.Table{
+				&executetest.Table{
+					KeyCols: []string{"tag0", "tag1"},
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "_value", Type: flux.TFloat},
+						{Label: "tag0", Type: flux.TString},
+						{Label: "tag1", Type: flux.TString},
+					},
+					Data: [][]interface{}{
+						{execute.Time(3600000000000 * 4), 2.0, "a", "b"},
+						{execute.Time(3600000000000 * 28), 2.0, "a", "c"},
+						{execute.Time(3600000000000 * 6), 2.0, "a", "b"},
+						{execute.Time(3600000000000 * 30), 2.0, "a", "d"},
+						{execute.Time(3600000000000 * 32), 2.0, "a", "d"},
```

Comment:
When you have different group key values, you should separate the input in multiple tables.